ui/src/pages/Edit/UploadStory.tsx

```diff
@@ -0,0 +1,721 @@
+  margin-right: 5px;
+  width: 2vw;
+  float: right;
+  height: 3.75vh;
+
+  .MuiButton-label {
+    color: ${colors.primaryDark1};
+    font-family: Poppins !important;
+  }
+`;
+const StyledBackgroundColor = styled.div`
+  background-color: ${colors.white};
+  width: 40vw;
+  padding: 0px 0px 24px 24px;
+  margin-bottom: 24px;
+`;
+
+const StyledLinearProgress = styled(LinearProgress)`
+  margin-top: 24px;
+`;
+
+const StyledImage = styled.img`
+  width: 30vw;
+`;
+
+export const UploadStory: React.FC<StoryProps> = ({
+  id,
+  currentStory,
+  bio,
+}: StoryProps) => {
+  const { data: tagOptions, error } = useSWR<string[]>("/api/tags");
+  const { data: countries, error: errorCountries } = useSWR<string[]>(
+    "/api/countries"
+  );
+  const [state, dispatch] = useReducer(
+    uploadStoryReducer,
+    INIT_STATE(currentStory)
+  );
+
+  const startYear = new Date().getFullYear();
+  const yearArray = Array.from({ length: 30 }, (_, i) => startYear - i);
+
+  const [formInput, setFormInput] = useReducer(
+    (state, newState) => ({ ...state, ...newState }),
+    {
+      image: new File([""], "filename"),
+      video_url: currentStory.video_url,
+      title: currentStory.title,
+      content: currentStory.content,
+      summary: currentStory.summary,
+      author_first_name: currentStory.author_first_name,
+      author_last_name: currentStory.author_last_name,
+      author_country: currentStory.author_country,
+      year: currentStory.year,
+      current_city: currentStory.current_city.toUpperCase(),
+      bio: bio,
+    }
+  );
+
+  const story = React.useMemo(() => {
+    if (state.isDrawerOpen) {
+      const storyFromData: Story = {
+        image_url:
+          state.newImage === ""
+            ? URL.createObjectURL(formInput.image)
+            : currentStory.image_url,
+        video_url: formInput.video_url as string,
+        title: formInput.title as string,
+        content: formInput.content as string,
+        summary: formInput.summary as string,
+        author_first_name: formInput.author_first_name as string,
+        author_last_name: formInput.author_last_name as string,
+        author_country: state.authorCountry as string,
+        year: formInput.year as number,
+        current_city: formInput.current_city as string,
+        author: {
+          ID: 0,
+          CreatedAt: "",
+          UpdatedAt: "",
+          DeletedAt: {
+            Time: "",
+            Valid: true,
+          },
+          firstName: formInput.author_last_name as string,
+          currentCity: formInput.current_city as string,
+          bio: formInput.bio as string,
+        } as Author,
+        tags: [],
+        is_visible: true,
+        latitude: 0,
+        longitude: 0,
+        ID: 0,
+        CreatedAt: "",
+        UpdatedAt: "",
+        DeletedAt: {
+          Time: "",
+          Valid: true,
+        },
+      };
+      return storyFromData;
+    }
+    return undefined;
+  }, [state.isDrawerOpen, formInput]);
```

Comment:
We should add the dependencies, otherwise we risk not invalidating the cache properly.